src/backend/config/settings.py

```diff
@@ -177,6 +177,15 @@ class GDriveSettings(BaseSettings, BaseModel):
         ),
     )
 
+class GoogleWebSearchSettings(BaseSettings, BaseModel):
+    model_config = SETTINGS_CONFIG
+    api_key: Optional[str] = Field(
+        default=None, validation_alias=AliasChoices("API_KEY", "api_key")
```

Comment:
The alias determines the key for the `.env` (we keep this logic to make env users backwards compatible). I would change the name here to something more explicit, like `GOOGLE_SEARCH_API_KEY`

Same thing for the `CSE_ID`